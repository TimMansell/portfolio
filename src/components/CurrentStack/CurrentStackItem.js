import React from 'react';
import PropTypes from 'prop-types';

import Image from '../Image';

import styles from './CurrentStack.module.scss';

export const CurrentStackItem = ({ url, name, imgWidth }) => {
  const img = name.toLowerCase();

  return (
    <div className={styles.item}>
      <a
        href={url}
        target="_blank"
        rel="noopener noreferrer"
        data-test="stack-item-link"
      >
        <Image
          src={`CurrentStack/img/${img}.svg`}
          alt={name}
          width={imgWidth}
        />
      </a>
    </div>
  );
};

CurrentStackItem.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  imgWidth: PropTypes.string.isRequired,
};

export default CurrentStackItem;
