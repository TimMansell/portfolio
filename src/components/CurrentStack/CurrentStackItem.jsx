import React from 'react';
import PropTypes from 'prop-types';

import Image from '../Image';

import styles from './CurrentStack.module.scss';

export const CurrentStackItem = ({ url, name, width, height }) => {
  const img = name.toLowerCase();

  return (
    <div className={styles.item}>
      <a
        href={url}
        target="_blank"
        rel="noopener noreferrer"
        data-test="stack-item-link"
      >
        <Image
          src={`CurrentStack/img/${img}.svg`}
          alt={name}
          width={width}
          height={height}
        />
      </a>
    </div>
  );
};

CurrentStackItem.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
};

export default CurrentStackItem;
