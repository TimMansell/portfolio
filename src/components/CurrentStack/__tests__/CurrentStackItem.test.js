import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CurrentStackItem from '../CurrentStackItem';

const props = {
  url: 'test url',
  name: 'babel',
  width: '50',
  height: '50',
};

describe('CurrentStackItem', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<CurrentStackItem {...props} />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<CurrentStackItem {...props} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should be correct link href', () => {
    const wrapper = shallow(<CurrentStackItem {...props} />);
    const link = wrapper.find('[data-test="stack-item-link"]');

    expect(link.props().href).toEqual(props.url);
  });
});
