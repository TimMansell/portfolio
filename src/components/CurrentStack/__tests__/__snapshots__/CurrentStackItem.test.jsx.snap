// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CurrentStackItem > should match snapshot 1`] = `
<div
  className="_item_2bf000"
>
  <a
    data-test="stack-item-link"
    href="test url"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      className="_container_4c41c3"
    >
      <img
        alt="babel"
        className="_image_4c41c3"
        loading="lazy"
        src="file:///H:/projects/portfolio/src/components/CurrentStack/img/babel.svg"
        title="babel"
        width="50"
      />
    </div>
  </a>
</div>
`;
